#VRML_SIM R2021b utf8
WorldInfo {
  info [
    "Autonomous Vehicle Simulation"
    "The simple controller example uses an on-board camera to follow the yellow road lines and a SICK sensor to avoid the obstacles."
    "The control of the vehicle is done using the driver library."
    "The vehicle based on the Car PROTO is modelled with realistic physics properties: motor torques, body mass, friction forces, suspensions, etc."
  ]
  title "City"
  ERP 0.6
  basicTimeStep 10
  coordinateSystem "NUE"
  lineScale 1
  contactProperties [
    ContactProperties {
      coulombFriction [
        8
      ]
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    ContactProperties {
      material2 "CitroenCZeroWheels"
      coulombFriction [
        8
      ]
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    ContactProperties {
      material2 "ToyotaPriusWheels"
      coulombFriction [
        8
      ]
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    ContactProperties {
      material2 "LincolnMKZWheels"
      coulombFriction [
        8
      ]
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    ContactProperties {
      material2 "RangeRoverSportSVRWheels"
      coulombFriction [
        8
      ]
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
    ContactProperties {
      material2 "TruckWheels"
      coulombFriction [
        8
      ]
      softCFM 1e-05
      bumpSound ""
      rollSound ""
      slideSound ""
    }
  ]
}
Viewpoint {
  orientation 0.13855075550165466 0.9485638481489291 0.28465824093958003 3.9593212421848505
  position -22.25070804030367 32.94458762007454 -15.13373847358087
  near 1
  followSmoothness 0
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF GROUND Solid {
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        baseColorMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2021b/projects/vehicles/worlds/textures/ground.jpg"
          ]
        }
        roughness 0.5
        metalness 0
        textureTransform TextureTransform {
          scale 500 500
        }
      }
      geometry DEF GROUND_PLANE Plane {
        size 2000 2000
      }
    }
  ]
  boundingObject USE GROUND_PLANE
  locked TRUE
}
StraightRoadSegment {
  translation 0 0.02 140
  rotation 0 1 0 3.1416
  name "road(3)"
  id "3"
  startJunction "23"
  endJunction "22"
  width 21.5
  numberOfLanes 4
  numberOfForwardLanes 2
  lines [
    RoadLine {
      width 0.3
    }
    RoadLine {
      color 0.85 0.75 0.3
      type "double"
    }
  ]
  length 160
}
BmwX5 {
  translation 3.27 0.4 12.42
  rotation 0 1 0 -5.307179586466759e-06
  color 0.933333 0.933333 0.92549
  name "robot_2"
  controller "vision_controller"
  sensorsSlotFront [
    SickLms291 {
      translation 0 -2.4286129e-17 -0.06
    }
  ]
  sensorsSlotTop [
    Display {
    }
    Camera {
      translation 0 -0.05 -0.72
      children [
        Transform {
          translation 0 0 0.05
          rotation 1 0 0 1.5708
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.8 0.8 0.8
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.1
                radius 0.04
              }
            }
          ]
        }
      ]
      fieldOfView 1
      width 128
      recognition Recognition {
        frameThickness 0
        segmentation TRUE
      }
    }
  ]
  sensorsSlotCenter [
    Emitter {
      name "emitter_AV2"
      channel 2
    }
    Compass {
      rotation 0 1 0 -3.141592653589793
      name "compass_AV2"
      resolution 0.01
    }
    GPS {
      translation 0 0 -3.1
      name "global_gps_AV2"
    }
    Gyro {
    }
    Display {
      width 200
      height 150
    }
  ]
}
Solid {
  translation -2.5 0 0
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Cylinder {
        height 0.2
        radius 0.6
      }
    }
  ]
  name "solid(2)"
}
Solid {
  translation -2.5 0 15
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Cylinder {
        height 0.2
        radius 0.6
      }
    }
  ]
  name "solid(3)"
}
Solid {
  translation -2.5 0 30
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Cylinder {
        height 0.2
        radius 0.6
      }
    }
  ]
  name "solid(4)"
}
Solid {
  translation -2.5 0 45
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Cylinder {
        height 0.2
        radius 0.6
      }
    }
  ]
  name "solid(5)"
}
Solid {
  translation -2.5 0 60
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Cylinder {
        height 0.2
        radius 0.6
      }
    }
  ]
  name "solid(6)"
}
Solid {
  translation 0 0 80
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Cylinder {
        height 0.2
        radius 0.6
      }
    }
  ]
  name "goal_0(2)"
}
SolidBox {
  translation 6.94 0 29.28
  appearance PBRAppearance {
    baseColor 1 0 0
    baseColorMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2021b/projects/default/worlds/textures/tagged_wall.jpg"
      ]
    }
    roughness 0.5
    metalness 0
  }
}
